cmd/app/main.go

```
@@ -0,0 +1,59 @@
+		muxHandler = middleware.RateLimiterMiddleware(limiter)(muxHandler)
+	}
+
+	// Initialize the response encoder
+	responseEncoder := http_response_encoder.NewJSONEncoder()
+	// Initialize handlers with dependency injection
+	apiHandler := handler.NewAPIHandler(responseEncoder)
+	helloHandler := handler.NewHelloHandler(responseEncoder)
+
+	mux.Handle("/api", apiHandler)
+	mux.Handle("/api/hello", helloHandler)
+	http.ListenAndServe(fmt.Sprintf(":%d", cfg.Server.Port), muxHandler)
```

golangci-lint (1.64.8)

Error return value of `http.ListenAndServe` is not checked

(errcheck)